.github/workflows/pre-ci.yml

```diff
@@ -18,9 +18,13 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: "3.8"
+          cache: "pip"
 
       - name: Install pre-commit
-        run: pip install pre-commit==3.2.2
-
+        run: python -m pip install -r pre-commit-requirements.txt
```

Comment:
why not pip directly but through module? required?

Reply:
Isn't it safer? You are guaranteed that you are using the correct pip/module